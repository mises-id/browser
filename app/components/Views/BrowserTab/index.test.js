/*
 * @Author: lmk
 * @Date: 2021-07-05 10:09:07
 * @LastEditTime: 2021-08-14 18:35:16
 * @LastEditors: lmk
 * @Description:
 */
jest.useFakeTimers();

import React from 'react';
import {shallow} from 'enzyme';
import {BrowserTab} from './';

describe('Browser', () => {
  it('should render correctly', () => {
    const navigation = {
      state: {},
    };
    const wrapper = shallow(
      <BrowserTab initialUrl="https://mises.site" navigation={navigation} />,
    );
    expect(wrapper).toMatchSnapshot();
  });
});
