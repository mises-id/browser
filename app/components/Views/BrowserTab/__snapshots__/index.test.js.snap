// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Browser should render correctly 1`] = `
<ErrorBoundary
  view="BrowserTab"
>
  <View
    style={
      Array [
        Object {
          "backgroundColor": "#FFFFFF",
          "flex": 1,
        },
        false,
      ]
    }
  >
    <View
      style={
        Object {
          "flex": 1,
          "zIndex": 1,
        }
      }
    />
    <View
      style={
        Object {
          "height": 3,
          "left": 0,
          "position": "absolute",
          "right": 0,
          "top": 0,
          "width": "100%",
          "zIndex": 999999,
        }
      }
    >
      <WebviewProgressBar
        progress={0}
      />
    </View>
    <ReactNativeModal
      animationIn="slideInDown"
      animationInTiming={300}
      animationOut="slideOutUp"
      animationOutTiming={300}
      avoidKeyboard={false}
      backdropColor="black"
      backdropOpacity={0.7}
      backdropTransitionInTiming={300}
      backdropTransitionOutTiming={300}
      coverScreen={true}
      customBackdrop={null}
      deviceHeight={null}
      deviceWidth={null}
      hasBackdrop={true}
      hideModalContentWhileAnimating={false}
      isVisible={false}
      onBackButtonPress={[Function]}
      onBackdropPress={[Function]}
      onModalHide={[Function]}
      onModalShow={[Function]}
      onModalWillHide={[Function]}
      onModalWillShow={[Function]}
      panResponderThreshold={4}
      propagateSwipe={false}
      scrollHorizontal={false}
      scrollOffset={0}
      scrollOffsetMax={0}
      scrollTo={null}
      style={
        Object {
          "justifyContent": "flex-start",
          "margin": 0,
        }
      }
      supportedOrientations={
        Array [
          "portrait",
          "landscape",
        ]
      }
      swipeThreshold={100}
      useNativeDriver={true}
    >
      <View
        style={
          Object {
            "backgroundColor": "#FFFFFF",
            "flexDirection": "row",
            "height": 87,
            "paddingHorizontal": 10,
            "paddingTop": 50,
          }
        }
        testID="url-modal"
      >
        <TextInput
          autoCapitalize="none"
          autoCorrect={false}
          clearButtonMode="while-editing"
          keyboardType="web-search"
          onChangeText={[Function]}
          onSubmitEditing={[Function]}
          placeholder="Search or Type URL"
          placeholderTextColor="#848c96"
          returnKeyType="go"
          selectTextOnFocus={true}
          style={
            Object {
              "backgroundColor": "#f2f3f4",
              "borderRadius": 30,
              "flex": 1,
              "fontFamily": "EuclidCircularB-Regular",
              "fontSize": 14,
              "fontWeight": "400",
              "height": 30,
              "padding": 8,
              "paddingLeft": 15,
              "textAlign": "left",
            }
          }
          testID="url-input"
          value=""
        />
        <TouchableOpacity
          onPress={[Function]}
          style={
            Object {
              "marginLeft": 10,
              "marginTop": 7,
            }
          }
          testID="ios-cancel-url-button"
        >
          <Text
            style={
              Object {
                "color": "#037dd6",
                "fontFamily": "EuclidCircularB-Regular",
                "fontSize": 14,
                "fontWeight": "400",
              }
            }
          >
            [missing "en.browser.cancel" translation]
          </Text>
        </TouchableOpacity>
      </View>
      <Connect(UrlAutocomplete)
        input=""
        onDismiss={[Function]}
        onSubmit={[Function]}
      />
    </ReactNativeModal>
    <BrowserBottomBar
      canGoBack={false}
      canGoForward={false}
      goBack={[Function]}
      goForward={[Function]}
      goHome={[Function]}
      showTabs={[Function]}
      showUrlModal={[Function]}
      toggleOptions={[Function]}
    />
  </View>
</ErrorBoundary>
`;
